discovery_engine_core/providers/src/client.rs

```diff
@@ -122,9 +141,10 @@ impl Client {
             .error_for_status()
             .map_err(Error::StatusCode)?;
 
-        let news: NewscatcherResponse = response.json().await.map_err(Error::Decoding)?;
-        let result: Vec<Article> = news.articles.into_iter().collect();
-        Ok(result)
+        let raw_response = response.text().await.map_err(Error::Fetching)?;
+        let deserializer = &mut serde_json::Deserializer::from_str(raw_response.as_str());
```

Comment:
this could be coerced automatically
```suggestion
        let deserializer = &mut serde_json::Deserializer::from_str(&raw_response);
```